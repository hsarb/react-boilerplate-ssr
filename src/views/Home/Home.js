// @flow
import React, { Component } from 'react';
import fetch from 'isomorphic-fetch';
import Helmet from 'react-helmet';
import Typography from 'material-ui/Typography';
import withStyles from 'material-ui/styles/withStyles';

const styles = {
  root: {
    alignItems: 'center',
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'column',
    minHeight: 400,
  },
};

type Props = {
  classes: Object,
  message: ?string,
};

class Home extends Component<Props> {
  static async getInitialProps() {
    const call = await fetch('http://localhost:3000/api');
    const { message } = await call.json();

    return { message };
  }

  render() {
    const { classes, message } = this.props;

    return (
      <div className={classes.root}>
        <Helmet title="Welcome" />
        <Typography variant="display1">{message}</Typography>
        <Typography variant="title">
          A minimal library React boilerplate built with performance, testing, and ease of use in
          mind.
        </Typography>
      </div>
    );
  }
}

export default withStyles(styles)(Home);
