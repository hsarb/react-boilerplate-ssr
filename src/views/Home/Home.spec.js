import Home from './Home';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Home', () => {
  it('should render correctly', () => {
    const props = {
      message: 'Hello world',
    };

    const wrapper = renderer.create(<Home {...props} />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
