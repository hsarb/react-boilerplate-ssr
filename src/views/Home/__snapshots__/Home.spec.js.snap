// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<div
  className="Home-root-1"
>
  <h1
    className="MuiTypography-root-2 MuiTypography-display1-6"
  >
    Hello world
  </h1>
  <h2
    className="MuiTypography-root-2 MuiTypography-title-8"
  >
    A minimal library React boilerplate built with performance, testing, and ease of use in mind.
  </h2>
</div>
`;
